src/app/filter/filter.model.ts

```diff
@@ -1,5 +1,11 @@
+import { FormControl } from '@angular/forms';
+
 export interface Filters {
   currentPage: number;
   entriesPerPage: number;
   nameFilter?: string;
 }
+
+export type FiltersFormGroup = {
+  [k in keyof Filters]: FormControl<Filters[k] | null>;
+};
```

Comment:
Record<> is made for this, IIRC

Reply:
https://stackoverflow.com/questions/51936369/what-is-the-record-type-in-typescript

Reply:
I can't use it in this case, because I'm not just mapping keys to a specific type.
Instead I'm enforcing that for each key of Filters the value is always going to be a FormControl, BUT of the same type of the value returned for that specific key of Filters or null.
While Record maps keys always to the same value type, regardless of which is the specific key.

Reply:
ok, good to know

Reply:
I was convinced that `Record<keyof Filters, FormControl<Filters[k] | null>>` it's the same